flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -121,6 +122,7 @@ private FailureHandlingResult handleFailure(
                     failedExecution,
                     new JobException("The failure is not recoverable", cause),
                     timestamp,
+                    FailureEnricherUtils.EMPTY_LABELS,
                     globalFailure);
         }
 
```

Comment:
How about to do failure labeling in this method, all failures of DefaultScheduler will go through this path. And extra benefits are:
1. FailureHandlingResult can host all kinds of labels, no matter it's global or task failure.
2. Can simplify the modification of Execution. No need to modify fail()/markFailed()/etc.
3. No need to change TaskExecutionStateTransition to temporarily host the labels.